spec/requests/authentication_controller_spec.rb

```diff
@@ -0,0 +1,70 @@
+RSpec.describe AuthenticationController do
+  before do
+    discovery_response = instance_double(
+      "OpenIDConnect::Discovery::Provider::Config::Response",
+      authorization_endpoint: "http://openid-provider/authorization-endpoint",
+      token_endpoint: "http://openid-provider/token-endpoint",
+      userinfo_endpoint: "http://openid-provider/userinfo-endpoint",
+      end_session_endpoint: "http://openid-provider/end-session-endpoint",
+    )
+
+    # rubocop:disable RSpec/AnyInstance
+    allow_any_instance_of(OidcClient).to receive(:discover).and_return(discovery_response)
```

Comment:
How come we need `allow_any_instance_of` here?
Is this to save us having to put an `allow()` in every block where we're defining an instance of this class with a `let`?

Reply:
The tests don't have access to the actual `OidcClient` instance because they're testing via the controllers.  An alternative way of doing this would be to use webmock to stub the discovery response, but that's kind of awkward because the `openid_connect` gem expects everything to be set up with proper crypto.